### ISSUE TITLE
Issues with Cowdata.h when compiling

### ISSUE THREAD

**Issue by Gramps** [godotengine/godot#62383]
### Godot version

3.4.4-stable

### System information

Windows 10 64-bit Latest

### Issue description

When compiling with:
 `scons platform=windows tools=yes target=release_debug production=yes`
I will get this warning:
`core\cowdata.h(136) : warning C4724: potential mod by 0`
Without the `production=yes flag`, the warning does not seem to appear.

When using flags that Godot's Github Actions uses, from what I can glean:
`scons platform=windows target=release_debug tools=yes verbose=yes warnings=all debug_symbols=no`
But adding the `production=yes` flag this warning becomes and error, halting compiling, obviously:
```
core\cowdata.h(136) : error C2220: the following warning is treated as an error
core\cowdata.h(136) : warning C4724: potential mod by 0
```

### Steps to reproduce

Basically attempt to compile with the flags above.  Compiling with MSVC version 14.2,

**Comment by akien-mga** [comment]
Note that the warnings are only treated as error if `werror=yes` is passed.

**Comment by Gramps** [comment]
Nope, that is all it tells me.  It has been doing this for a while, perhaps as far back as Godot 3.4.2; I would have assumed it got mentioned but I couldn't find any open issues with it.

**Comment by RicardRC** [comment]
I got more info on this:
```
...\godot\core\templates\cowdata.h(199) : error C2220: the following warning is treated as an error
...\godot\core\templates\cowdata.h(199) : warning C4724: potential mod by 0
...\godot\core\templates\cowdata.h(199) : warning C4724: potential mod by 0
scons: *** [editor\plugins\gizmos\collision_polygon_3d_gizmo_plugin.windows.editor.double.x86_64.obj] Error 2
```

**Comment by RicardRC** [comment]
This is on 4.4.1 release tag, btw. Looking at `godot/editor/plugins/gizmos/collision_polygon_3d_gizmo_plugin.cpp` : Lines 100, 123, 137, 138, 139, 168, 179, 208 all could trigger this warning, they have modulus and .size() at the denominator. But they all follow this pattern or equivalents, so it can never actually be zero for any of them:

>    for (int i = 0; i < points.size(); i++) {
>        int n = (i + 1) % points.size();

**Comment by AThousandShips** [comment]
I think this has been solved in 4.x, please check latest 4.5 branch

**Comment by RicardRC** [comment]
Ok, I just switched to master branch latest commit and recompiled a clean source tree. It appears solved in 4.5 indeed.
